backend/remote-state/azure/client.go

```diff
@@ -39,7 +40,7 @@ func (c *RemoteClient) Get() (*remote.Payload, error) {
 	ctx := context.TODO()
 	blob, err := c.giovanniBlobClient.Get(ctx, c.accountName, c.containerName, c.keyName, options)
 	if err != nil {
-		if blob.Response.StatusCode == 404 {
```

Comment:
`blob.Response` is an `autorest.Response`, which points to an embedded `*http.Response`. While the `autorest` method puts in `nil` pointer check to help prevent panics, I would prefer this to not access an invalid return value in the first place. If there was an error, there is no reason to check for a 404, since in the case of an error there is no http response to check for a status.

Reply:
Azure returns an error when the status codes don't match what they expect, so we need to check for the 404 here since an error will be returned but not as useful to users unless we explicitly check the status code, unfortunately

Reply:
Thanks @tombuildsstuff, I didn't realize azure work contrary to all other Go APIs, go ahead and ignore my review ;)